cli/localci/core/boost_cache.py

```
@@ -0,0 +1,97 @@
+    try:
+        subprocess.run(args, check=True, capture_output=True, text=True)
+        logger.info("Boost cache cloned at %s (branch=%s)", dest, branch)
+    except subprocess.CalledProcessError as e:
+        logger.warning(
+            "Boost cache clone failed: %s (stderr: %s)",
+            e,
+            (e.stderr or "").strip() or "(none)",
+        )
+
+
+def _git_fetch_and_update(dest: Path, branch: str, shallow: bool = False) -> None:
```

Ruff (0.15.2)

[warning] Unused function argument: `shallow`

(ARG001)